### PR TITLE
Add slash handling and command format handling

### LINKED ISSUE #123: [PE-D][Tester E] The guide showed incorrect error
I tried the add command with the wrong prefix for tag with tag/ instead of t/ but the error shows that there was an issue with role but not tag instead with "Invalid role specified. Must be one of: client, employee, supplier." This is misleading and does not help users to address what was wrong with their command.

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: wolffe88/ped#4

### LINKED ISSUE #170: [PE-D][Tester E] Non relevent guide for edit command
The error should show an invalid command format instead of stating "Ids should only be any integer value larger than 0" for which I have input correctly

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: wolffe88/ped#15

### LINKED ISSUE #218: Incorrect errors when the user typing in an incorrect prefix
Previously when an incorrect prefix is entered, the app either just accept it to be an input for previous prefix or just say that that the input is incorrectly formatted.

Expecting the app to warn when the user unintentionally typing in an incorrect prefix

### LINKED ISSUE #219: The incorrect input format is not shown when adding a person
Previously the app do nothing when a user adding an incorrect input format.

Expecting the app to show the correct error to response to the user input / instructions.

### PR DESCRIPTION
**Pull request by purivirakarin**
Close #218, close #123, close #170 (Incorrect prefix string warning)

Close #219 (Refactor error handling for AddCommandParser)

### CODE REVIEW

File: src/main/java/seedu/address/model/util/TestCommandFormatUtil.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package seedu.address.model.util;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Utility class for testing command format validation.
+ */
+public class TestCommandFormatUtil {
+    public static final String MESSAGE_CONSTRAINTS = "Invalid command format! Invalid prefix has been detected.";
```
Comment (shaunlxw) [inline src/main/java/seedu/address/model/util/TestCommandFormatUtil.java]: `"Invalid command format! \n%1$s"` already exists in Messages.java. Could possibly use that and keep this as just `Invalid prefix has been detected.` to add on to the string, for consistency.

**Review by shaunlxw — CHANGES_REQUESTED** [review]
I think we need to separately do a slash check for role in add command parser, as the following command does not output the "Invalid prefix" message (since we do not have a parseRole).

`add n/john p/123 e/john@example.com address/blk 123 serangoon role/client tag/friend`

File: src/main/java/seedu/address/model/util/TestCommandFormatUtil.java (hunk 1 above)
Comment (purivirakarin) [reply src/main/java/seedu/address/model/util/TestCommandFormatUtil.java]: Thanks. Resolved!

**Review by shaunlxw — APPROVED** [review]
LGTM! 👍 Can consider adding more tests to improve codecov, or else can be merged! Opened #220 as continuation.

Edit: should we consider just using the same "Invalid command format! ..." message for any invalid prefix found? Currently, it might be inconsistent.

input: `delete name/john` it shows "invalid command format! ..."
input: `delete n/john name/roy` it shows "invalid prefix has been detected".

**Pull request merged**